examples/connext_dds/instance_statistics/c++/instance_publisher.cxx

```diff
@@ -0,0 +1,214 @@
+    if (retcode != DDS_RETCODE_OK) {
+        return shutdown_participant(participant, "register_type error", EXIT_FAILURE);
+    }
+
+    // Create a Topic with a name and a datatype
+    DDSTopic *topic = participant->create_topic(
+        "Example instance",
+        type_name,
+        DDS_TOPIC_QOS_DEFAULT,
+        NULL, // listener
+        DDS_STATUS_MASK_NONE);
+    if (topic == NULL) {
+        return shutdown_participant(participant, "create_topic error", EXIT_FAILURE);
+    }
+
+    // This DataWriter writes data on "Example instance" Topic
+    DDSDataWriter *untyped_writer = publisher->create_datawriter(
+        topic,
+        DDS_DATAWRITER_QOS_DEFAULT,
+        NULL, // listener
+        DDS_STATUS_MASK_NONE);
+    if (untyped_writer == NULL) {
+        return shutdown_participant(participant, "create_datawriter error", EXIT_FAILURE);
+    }
+
+    // Narrow casts from an untyped DataWriter to a writer of your type 
+    instanceDataWriter *typed_writer =
+    instanceDataWriter::narrow(untyped_writer);
+    if (typed_writer == NULL) {
+        return shutdown_participant(participant, "DataWriter narrow error", EXIT_FAILURE);
+    }
+
+    // Create data for writing, allocating all members
+    instance *data = instanceTypeSupport::create_data();
+    if (data == NULL) {
+        return shutdown_participant(
+            participant,
+            "instanceTypeSupport::create_data error",
+            EXIT_FAILURE);
+    }
+
+    // Main loop, write data
+    DDS_InstanceHandle_t instance_handle = DDS_HANDLE_NIL;
+    DDS_DataWriterCacheStatus status;
+    for (unsigned int samples_written = 0;
+    !shutdown_requested && samples_written < sample_count;
+    ++samples_written) {
+
+        // Modify the data to be written here
+        data->x = static_cast<DDS_Short>(samples_written);
+
+        std::cout << "Writing instance, count " << samples_written 
+        << std::endl;
+        instance_handle = typed_writer->register_instance(*data);
+        retcode = typed_writer->write(*data, instance_handle);
+        if (retcode != DDS_RETCODE_OK) {
+            std::cerr << "write error " << retcode << std::endl;
+        }
+
+        // Send once every second
+        DDS_Duration_t send_period = { 1, 0 };
+        NDDSUtility::sleep(send_period);
+        if (samples_written % 2 == 0) {
+            // Unregister the instance
+            retcode = typed_writer->unregister_instance(
+                    *data, instance_handle);
+            if (retcode != DDS_RETCODE_OK) {
+                std::cerr << "register_instance error " << retcode << std::endl;
+            }
+        } else if (samples_written % 3 == 0) {
+            // Dispose the instance
+            retcode = typed_writer->dispose(
+                    *data, instance_handle);
+            if (retcode != DDS_RETCODE_OK) {
+                std::cerr << "dispose error " << retcode << std::endl;
+            }
```

Comment:
in both above cases should shutdown on error I think

Reply:
It could be one possibility. However, I consider we should log the error and continue the same approach as we do in the write operation.